fix: generic provider format

zackhee997 opened the pull request:
1. fix assume role block format
2. missing region

zackhee997 committed:
fix: generic provider roleARN format

zackhee997 committed:
fix: generic provider missing region

vincenthsh committed:
chore: Add docs and fix testdata

vincenthsh commented on testdata/generic_providers_yaml/fogg.yml:
```diff
@@ -48,7 +48,8 @@ envs:
-          aws_assume_role: "TerraformExecutionRole"
+          assume_role: "TerraformExecutionRole"
+          session_name: "foo"
     components:
```
as suggested by @zackhee997  - should we instead expect

```yaml
assume_role:
  role: "TerraformExecutionRole"
  session_name: ...
```

and resolver special case to combine `role` value with `AwsConfig` accountID / ... to auto generate `role_arn`?

vincenthsh committed:
chore: Refactor generic provider assume_role map support

- Align closer to TF assume role block to expect map in both cases
- Add genericProvider validation to component validation
- Add tests for genericProvider validation function

vincenthsh committed:
chore: don't override region if provided
